File: Python3/1047. Remove All Adjacent Duplicates In String.py
class Solution:
    def removeDuplicates(self, S: str) -> str:
        # while(re.search(r'([a-z])\1',S)):
        #     S = re.sub(r'([a-z])\1','',S)
        # return S
        stack = []
        for i in S:
            if stack!=[] and i == stack[-1]:
                stack.pop()
            else:
                stack.append(i)
        return ''.join(stack)

File: C++/1047. Remove All Adjacent Duplicates In String.cpp
class Solution {
public:
    string removeDuplicates(string S) {
        string ss;
        for(auto c: S){
            if(!ss.empty() && c==ss.back()){
                ss.pop_back();
            }
            else{
                ss.push_back(c);
            }
        }
        return ss;
    }
};
